src/main/java/frc/robot/subsystems/Arm.java

```diff
@@ -0,0 +1,155 @@
+package frc.robot.subsystems;
+
+import com.revrobotics.spark.SparkBase.PersistMode;
+import com.revrobotics.spark.SparkBase.ResetMode;
+import com.revrobotics.spark.SparkFlex;
+import com.revrobotics.spark.SparkLimitSwitch;
+import com.revrobotics.spark.SparkLowLevel.MotorType;
+import com.revrobotics.spark.config.SparkBaseConfig.IdleMode;
+import com.revrobotics.spark.config.SparkFlexConfig;
+import com.revrobotics.spark.config.SparkMaxConfig;
+
+import au.grapplerobotics.LaserCan;
+import edu.wpi.first.math.MathUtil;
+import edu.wpi.first.math.controller.PIDController;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.constants.ArmConstants;
+import frc.robot.constants.NeoMotorConstants;
+
+public class Arm extends SubsystemBase {
+    public enum ArmSetpoint {
+        L1Coral(0.0),
+        L2Coral(0.0),
+        L3Coral(0.0),
+        L4Coral(0.0),
+        L23Algae(0.0),
+        L34Algae(0.0),
+        Processor(0.0),
+        GroundAlgae(0.0),
+        Storage(0.0),
+        FeederStation(0.0);
+
+        public final double position;
+
+        ArmSetpoint(double position) {
+            this.position = position;
+        }
+
+        public double getPosition() {
+            return position;
+        }
+    }
+
+    private SparkFlex armMotor = new SparkFlex(ArmConstants.kArmMotorPort, MotorType.kBrushless);
+    private PIDController armController = new PIDController(ArmConstants.kP, ArmConstants.kI, ArmConstants.kD);
+    private SparkLimitSwitch lowerLimitSwitch = armMotor.getForwardLimitSwitch();
+    private LaserCan laser = new LaserCan(ArmConstants.kArmLaserMotorPort);
+    private static Arm instance;
+
+    /**
+     * Creates an instance of the Arm subsystem if it does not exist.
+     *
+     * @return An instance of the arm subsystem
+     */
+    public static Arm getInstance() {
+        if (instance == null) {
+            instance = new Arm();
+        }
+        return instance;
+    }
+
+    private Arm() {
+        SparkFlexConfig armConfig = new SparkFlexConfig();
+        armConfig.idleMode(IdleMode.kBrake)
+                .smartCurrentLimit(NeoMotorConstants.kMaxVortexCurrent);
+
+        armMotor.configure(armConfig, ResetMode.kResetSafeParameters,
+                PersistMode.kPersistParameters);
+
+        armController.setTolerance(ArmConstants.kTolerance);
+        armController.enableContinuousInput(0, ArmConstants.kEncoderConversion);
```

Comment:
This should be disable continuous input. Continuous input means your mechanism goes in a circle